src/benchmark.py

```diff
@@ -0,0 +1,235 @@
+import datetime
+import pytz
+import boto
+import boto3
+import s3fs
+import pandas as pd
+import json
+import psycopg2
+import sqlalchemy as sa # Package for accessing SQL databases via Python
+import StringIO
+
+def pd_to_postgres(df, table_name, mode):
+    """
+    Save DataFrame to PostgreSQL by to_sql() function.
+    """
+    # Writing Dataframe to PostgreSQL and replacing table if it already exists
+    df.to_sql(name=table_name, con=engine, if_exists = mode, index=False)
+
+def cleanColumns(columns):
+    cols = []
+    for col in columns:
+        col = col.replace(' ', '_')
+        cols.append(col)
+    return cols
+
+def df_to_sql(df, table_name, mode, new_table, psql, cur, engine):
+    """
+    Save DataFrame to .csv, read csv as sql table in memory and copy the table
+     directly in batch to PostgreSQL or Redshift.
+    """
+    # Prepare schema for table
+    df[:0].to_sql('temp', engine, if_exists = "replace", index=False)
+    df.columns = cleanColumns(df.columns)
+    # replace mode
+    if mode == 'replace':
+        cur.execute("DROP TABLE " + table_name)
+        cur.connection.commit()
+    # Prepare data to temp
+    if psql == 'psql':
+        data = StringIO.StringIO()
+        df.to_csv(data, header=False, index=False)
+        data.seek(0)
+        sql_query = """
+            COPY temp FROM STDIN WITH
+                CSV
+                HEADER;
+        """
+        cur.copy_expert(sql=sql_query, file=data)
+        cur.connection.commit()
+        # Copy or append table temp to target table
+        if (mode == 'replace' or new_table == 0):
+            sql_query2 = """
+                ALTER TABLE temp
+                RENAME TO %s;
+            """
+        else:
+            sql_query2 = """
+                INSERT INTO %s SELECT * FROM temp;
+                DROP TABLE temp;
+            """
+        cur.execute(sql_query2 % table_name)
+        cur.connection.commit()
+    else:
+        with s3.open('rxminer/temp.csv', 'wb') as f:
+            df.to_csv(f, index=False, header=False)
+        sql_query = """
+            COPY temp
+            FROM 's3://rxminer/temp.csv'
+            IAM_ROLE 'arn:aws:iam::809946175142:role/RedshiftCopyUnload'
+            CSV
+            IGNOREHEADER 1;
+            COMMIT;
+        """
+        cur.execute(sql_query)
+        # Copy or append table temp to target table
+        if (mode == 'replace' or new_table == 0):
+            sql_query2 = """
+                ALTER TABLE temp
+                RENAME TO %s;
+                COMMIT;VACUUM;COMMIT;
+            """
+        else:
+            sql_query2 = """
+                INSERT INTO %s SELECT * FROM temp;
+                DROP TABLE temp;
+                COMMIT;VACUUM;COMMIT;
+            """
+        cur.execute(sql_query2 % table_name)
+
+def read_medicaid(year, mode):
+    type_dir = 'sdud/medicaid_sdud_'
+    table_name = 'sdudtest'
+    cols_to_keep = [1,5,7,9,10,11,12,13,19]
+    column_names = [
```

Comment:
If you do what I suggested above, you would not need this list. Also managing a single dict is better than maintaining 2 lists.